### PR TITLE
fixed bugs that CNN was training during eval. Fixed bug that CNN was …

### PR DESCRIPTION
**Pull request by in1ght**
…not working due to new variables, small adjustments.

### CODE REVIEW

File: detector.py [hunk 1]
```diff
@@ -314,32 +314,33 @@ def main():
...
 
-        files_for_training = files_train + files_extra_train
+        files_for_training = files_train # + files_extra_train
 
 
         dict_extra = read_data(train_extra_dir)
         dict_train = read_data(train_dir)
         GT.update(dict_extra)
-        GT.update(dict_train)
+        # GT.update(dict_train)
```
Comment (DiMaths) [inline detector.py]: Why use train and train extra constants if we have parsed input dir as options.indir?
If you want to train on both folders it's either to actually create 3rd one on your machine called anyhow you like and then just use python3 ./detector.py ./combined_folder/ ./predictions_for_server.json ...

File: melspect_CNN.py
Comment (in1ght) [inline melspect_CNN.py:1]: that sets up in cmd, but the CNN needs both training and evaluating (i.e. not training). This small remark makes system be in a training mode forever and hence will lead to poor results and error if we use test for evaluating.

File: melspect_CNN.py [hunk 2]
```diff
@@ -213,7 +218,7 @@ def train_model(
     print("Working on device",torch.cuda.get_device_name(0))
```
Comment (DiMaths) [inline melspect_CNN.py]: If CUDA is not installed it crashes. I suggest switch default to no CUDA and using cpu and if CUDA is available then use it or even add --cuda option for potential usage with cpu when CUDA is available...

File: melspect_CNN.py [hunk 3]
```diff
@@ -213,7 +218,7 @@ def train_model(
...
 
     
-    optimizer = torch.optim.SGD(network.parameters(), lr=0.05, momentum=0.7)
+    optimizer = torch.optim.SGD(network.parameters(), lr=0.04, momentum=0.75)
```
Comment (DiMaths) [inline melspect_CNN.py:224]: Any reason why using SGD and not something more generic like Adam?

File: melspect_CNN.py [hunk 4]
```diff
@@ -129,20 +128,21 @@ def get_np_data(self,filename):
...
 
             ground_truth_onsets = np.zeros(melspects_3.shape[2])
-            ground_truth_onsets[onsets_ground_t_3] = 1
+            ground_truth_onsets[onsets_ground_t] = 1
+
+            for i in range(0,melspects_3.shape[2]-1):
+                if ground_truth_onsets[i] == 1:
+                    # working under assumption that onset is not fisrt or last element
+                    # and that there cannot be 2 frames with onsets in a row
+                    ground_truth_onsets[i-1] = 0.25
```
Comment (DiMaths) [inline melspect_CNN.py:144]: What 0.25 means?

**Review by DiMaths — CHANGES_REQUESTED** [review]

File: melspect_CNN.py
Comment (DiMaths) [reply melspect_CNN.py:1]: Sorry, now got it, thought it was redundant at first glance. I'll change it back in new commit

Comment (DiMaths) [reply melspect_CNN.py:1]: Changed it back. But do we treat dataset_val with is_training?

**Comment by DiMaths** [comment]
Would be nice if you could add the seed to the model and data handling functions. For reproducibility of splits, initialisations etc... I can make it a parser argument with default value later.

File: melspect_CNN.py
Comment (in1ght) [reply melspect_CNN.py:1]: yes, when we talk about dataset, since there is a ground truth for it, but as a model - no, in print_eval it is automatically stated that it treats model as is_training = False.

There are 2 Classes for it: 
1. Dataset (i.e. is_training = True, then in get_item we also give the ground truth for eval and train purposes)
2. Model (for dropout)

File: detector.py (hunk 1 above)
Comment (in1ght) [reply detector.py]: that is for convenience during working with data that has ground truth. You don't wanna model to train on train and then eval on train. This could be indeed done the way you suggested.

File: melspect_CNN.py (hunk 3 above)
Comment (in1ght) [reply melspect_CNN.py:224]: because it was used in one of the paper I read, and because of constast momentum than was a variable. Adam indeed could work better here. I initially worked with AdamW, but did not compare 2.

File: melspect_CNN.py (hunk 4 above)
Comment (in1ght) [reply melspect_CNN.py:144]: to smooth the model output it was suggested by Jan to give onset value of 0.25 to the neighboring frames next to the real onset.

**Comment by in1ght** [comment]
Currently working on SLIM RecSys, but will do when I am finished.

File: melspect_CNN.py
Comment (DiMaths) [reply melspect_CNN.py:1]: Ok, agreed.

File: detector.py (hunk 1 above)
Comment (DiMaths) [reply detector.py]: Now dict_train is never used and actually train files seem to be too, so conveniency is to comment out one out of 2 "folders" depending on which you want to train on and on which to evaluate?
But isn't evaluation done on the test split of files from the same folder? I find it more than fine if the model trains and evaluates on files from the same folder, but different ones.

Comment (in1ght) [reply detector.py]: it would require rewriting partially evaluate function, hence I used it different way, but that indeed makes sense. I will fix it on Monday if that works for you.

File: melspect_CNN.py (hunk 2 above)
Comment (DiMaths) [reply melspect_CNN.py]: Fixed
